pkg/sql/app_stats.go

```diff
@@ -230,20 +241,24 @@ func dumpStmtStats(ctx context.Context, appName string, stats map[string]*stmtSt
 // StmtStatsResetFrequency is the frequency at which per-app and
 // per-statement statistics are cleared from memory, to avoid
 // unlimited memory growth.
-var StmtStatsResetFrequency = envutil.EnvOrDefaultDuration(
-	"COCKROACH_SQL_STMT_STATS_RESET_INTERVAL", 1*time.Hour,
+var StmtStatsResetFrequency = settings.RegisterDurationSetting(
+	"sql.metrics.stmtstats.interval",
+	"frequency at which per-app and per-statement statistics are reset",
+	1*time.Hour,
 )
 
 // startResetWorker ensures that the data is removed from memory
 // periodically, so as to avoid memory blow-ups.
 func (s *sqlStats) startResetWorker(stopper *stop.Stopper) {
 	ctx := log.WithLogTag(context.Background(), "sql-stats", nil)
 	stopper.RunWorker(ctx, func(ctx context.Context) {
-		ticker := time.NewTicker(StmtStatsResetFrequency)
```

Comment:
because of the issues with stdlib timer in general? or just for the fixed-rate-ness of Ticker?